src/js/components/devices/terminal.js

```diff
@@ -136,12 +139,17 @@ export const Terminal = props => {
         if (obj.hdr.typ === MessageTypeNew) {
           if ((obj.hdr.props || {}).status == 2) {
             setSnackbar('Error: ' + byteArrayToString(obj.body), 5000);
+            snackbarAlreadySet = true;
+            socket.close();
             onCancel();
           } else {
             setSessionId(obj.hdr.sid);
           }
         } else if (obj.hdr.typ === MessageTypeShell) {
           term.write(byteArrayToString(obj.body));
+        } else if (obj.hdr.typ === MessageTypeStop) {
```

Comment:
Maybe this should be a "catch all" (else) case?

Reply:
uhm, not really: we are at risk of closing the socket if we do not recognize a message...

Reply:
Maybe this one is safer: https://github.com/mendersoftware/gui/pull/1206/commits/78a203f06a82fd61fa43cef92b2e1e5f85375c34